src/NServiceBus.Core.Analyzer/Sagas/RewriteConfigureHowToFindSagaFixer.cs

```diff
@@ -0,0 +1,194 @@
+
+            var title = diagnostic.Properties["_FixerTitle"];
+            var correlationId = diagnostic.Properties["_CorrelationId"];
+            var mapperParamName = diagnostic.Properties["_MapperParamName"];
+            var mappingCount = int.Parse(diagnostic.Properties["_MappingCount"]);
+            var configureHowToFindLocationStart = int.Parse(diagnostic.Properties["_ConfigureHowToFindLocation"]);
+
+            var configureMethodToken = root.FindToken(configureHowToFindLocationStart);
+            var method = configureMethodToken.Parent.AncestorsAndSelf().OfType<MethodDeclarationSyntax>().First();
+
+            var mappings = Enumerable.Range(0, mappingCount)
+                .Select(i =>
+                {
+                    var by2Index = i * 2;
+                    var msgTypeLocation = diagnostic.AdditionalLocations[by2Index];
+                    var mappingExpressionLocation = diagnostic.AdditionalLocations[by2Index + 1];
+                    var isHeader = bool.Parse(diagnostic.Properties[i.ToString()]);
+
+                    var messageType = root.FindToken(msgTypeLocation.SourceSpan.Start)
+                        .Parent.AncestorsAndSelf().OfType<IdentifierNameSyntax>().First();
+                    var mappingExpression = root.FindToken(mappingExpressionLocation.SourceSpan.Start)
+                        .Parent.AncestorsAndSelf().OfType<ArgumentSyntax>().First();
+
+                    return new SagaMessageMapping(0, messageType, null, isHeader, true, mappingExpression, null);
+                });
+
+            if (diagnostic.Properties.TryGetValue("_NewMappingForLocation", out var newMappingForLocation))
+            {
+                var baseTypeSyntax = root.FindToken(diagnostic.Location.SourceSpan.Start).Parent.AncestorsAndSelf().OfType<BaseTypeSyntax>().First();
+                var messageType = (baseTypeSyntax.Type as GenericNameSyntax).TypeArgumentList.Arguments[0];
+                var newMapping = SagaMessageMapping.CreateNewMapping(messageType, newMappingForLocation, correlationId);
+                mappings = mappings.Concat(new[] { newMapping });
+            }
+
+            // Register a code action that will invoke the fix
+            var action = CodeAction.Create(
+                title,
+                cancellationToken => RewriteConfigureHowToFindSagaMethod(context.Document, method, correlationId, mapperParamName, mappings.ToImmutableArray(), cancellationToken),
+                EquivalenceKey);
+
+            context.RegisterCodeFix(action, diagnostic);
+        }
+
+        static async Task<Document> RewriteConfigureHowToFindSagaMethod(
+            Document document,
+            MethodDeclarationSyntax method,
+            string correlationId,
+            string mapperParamName,
+            ImmutableArray<SagaMessageMapping> mappings,
+            CancellationToken cancellationToken)
+        {
+            // default(SyntaxNode).ToString() == empty string ""
```

Comment:
commented out code during development

Reply:
A note for the following lines. I'll change to:
```suggestion
            // Note: default(SyntaxNode).ToString() == empty string "", so you can compare the OrDefault value to string.
```